src/Hpack/Config.hs

```diff
@@ -467,6 +467,12 @@ toSection a CommonOptions{..}
     cppOptions = fromMaybeList commonOptionsCppOptions
     dependencies = fromMaybeList commonOptionsDependencies
 
+moduleNameFromPkgName :: String -> String
+moduleNameFromPkgName name = map f name
+  where
+      f '-' = '_'
+      f x = x
```

Comment:
Let's rename this to `pathsModuleFromPackageName` and generate the whole package name here.